src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,145 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
-import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { PageTitle } from './PageTitle';
+import { getPeople } from '../api';
+import { Person } from '../types';
 
 export const PeoplePage = () => {
+  const [data, setData] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [allPeople, setAllPeople] = useState<Person[]>([]);
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || 'asc';
+  const centuriesParam = searchParams.getAll('centuries');
+  const centuries = centuriesParam.length > 0 ? centuriesParam : [];
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        setIsLoading(true);
+        const people = await getPeople();
+
+        setData(people);
+        setAllPeople(people);
+      } catch (fetchError) {
+        setError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+
+    fetchData();
+  }, []);
+
+  const filteredData = (peopleData: Person[]) => {
```

Comment:
looks like you can pass more arguments (centuries, query, ..) and move this function into separated file src/helpers/filter-data.ts